### PR TITLE
[4.0] Send test mail as custom element

### PR DESCRIPTION
**Pull request by dgrammatiko**
Pull Request for Issue # .

### Summary of Changes
- move the code to custom element
- fix the session bug
~- remove the relative tests (will be added later on once we finish the first round of fields conversion to CE)~
- use namespaced classes in all the layouts (admin side only)

### Testing Instructions
Go to Admin Global Configuration and test the send mail functionality

## Preview
screen shot 2018-01-21 at 19 39 17

### Documentation Changes Required
No

### CODE REVIEW

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on 6f414c83302f6a41163bc9b80af69c3110321ecc<br /><br />Test using all 3 Mailers.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19426">issues.joomla.org/tracker/joomla-cms/19426</a>.</sub>

**Comment by anuragteapot** [comment]
@dgt41 did this PR also fix this issue?

screenshot from 2018-01-22 23-28-48

This issue occurs when a new user is created. 
Here also there should be two warning.

**Comment by dgrammatiko** [comment]
@Anu1601CS no, that issue is not related to front end, it’s php related

File: administrator/components/com_config/tmpl/application/default_mail.php [hunk 1]
```diff
@@ -7,28 +7,41 @@
  * @license     GNU General Public License version 2 or later; see LICENSE.txt
  */
 
+use \Joomla\CMS\Layout\LayoutHelper;
+use \Joomla\CMS\HTML\HTMLHelper;
```
Comment (anuragteapot) [inline administrator/components/com_config/tmpl/application/default_mail.php]: I think namespace should be written as  <code>Joomla\CMS\Layout\LayoutHelper</code>

Comment (dgrammatiko) [reply administrator/components/com_config/tmpl/application/default_mail.php]: Interesting, these use statements were automatically created by PHPStorm

Comment (anuragteapot) [reply administrator/components/com_config/tmpl/application/default_mail.php]: Oops

**Comment by anuragteapot** [comment]
@dgt41 can, I enable mail function and how?

**Comment by dgrammatiko** [comment]
@Anu1601CS it’s not joomla function it’s the server, yes you can enable it but the instructions to do so are very specific on your AMP server

**Comment by anuragteapot** [comment]
@dgt41  Thanks for the information.

**Comment by anuragteapot** [comment]
I have tested this item :white_check_mark: successfully on e2c6f55605e31f197a909cee5ba951472b36904e<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19426">issues.joomla.org/tracker/joomla-cms/19426</a>.</sub>

**Comment by ghost** [comment]
I have tested this item :red_circle: unsuccessfully on e2c6f55605e31f197a909cee5ba951472b36904e<br /><br />After PR applied click on Button "Send Test Mail" got no Info- or Error-Message; no Message in Console but:
bildschirmfoto 2018-01-24 um 08 14 39

<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19426">issues.joomla.org/tracker/joomla-cms/19426</a>.</sub>

**Comment by anuragteapot** [comment]
@franz-wohlkoenig please check on manual install of a patch.

**Comment by ghost** [comment]
@Anu1601CS what is meant by "manual install"?

**Comment by anuragteapot** [comment]
Just download this patch and install. Because it's working for me, that's why I told you to check by manual install.

**Comment by ghost** [comment]
I have tested this item :white_check_mark: successfully on e2c6f55605e31f197a909cee5ba951472b36904e

How to: 

- go to https://issues.joomla.org/tracker/joomla-cms/19426, 
- click on
bildschirmfoto 2018-01-24 um 09 06 36
- install and test
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19426">issues.joomla.org/tracker/joomla-cms/19426</a>.</sub>

**Comment by ghost** [comment]
Ready to Commit after two successful tests.

**Comment by dgrammatiko** [comment]
@wilsonge your call

**Comment by wilsonge** [comment]
@dgt41 next time can we keep the namespacing changes separate - makes reviewing 100x harder trying to separate things out

**Pull request merged**